### DISCUSSION TITLE
Error: Cannot find module Express

### DISCUSSION THREAD

**Discussion by matheusfenolio** [vercel/vercel#4649] [Help] [answered]
My repository: https://github.com/matheusfenolio/nlw-server

api url: https://nlw-server.matheusfenolio.vercel.app/

I created a folder called DIST and build the TypeScript there with yarn tsc. I did it before with another api and it worked, bur now I'm receiving this log:

19:34:35.234  
Retrieving list of deployment files...
19:34:36.551  
Downloading 11 deployment files...
19:34:37.687  
Installing build runtime...
19:34:39.533  
Build runtime installed: 1845.962ms
19:34:39.866  
Looking up build cache...
19:34:39.908  
Build cache found. Downloading...
19:34:39.917  
Build cache downloaded [305.00 B]: 8.031ms
19:34:40.253  
downloading user files...
19:34:40.253  
installing dependencies for user's code...
19:34:40.468  
yarn install v1.22.4
19:34:40.480  
info No lockfile found.
19:34:40.484  
[1/4] Resolving packages...
19:34:40.485  
[2/4] Fetching packages...
19:34:40.485  
[3/4] Linking dependencies...
19:34:40.490  
[4/4] Building fresh packages...
19:34:40.492  
success Saved lockfile.
19:34:40.493  
Done in 0.03s.
19:34:40.500  
running user script...
19:34:40.501  
preparing lambda files...
19:34:40.501  
compiling entrypoint with ncc...
19:34:40.721  
ncc: Version 0.18.5
19:34:40.721  
ncc: Compiling file index.js
19:34:40.871  
ncc: Module directory "/vercel/71ab9df5/src" attempted to require "celebrate" but could not be resolved, assuming external.
19:34:40.872  
ncc: Module directory "/vercel/71ab9df5/src" attempted to require "cors" but could not be resolved, assuming external.
19:34:40.872  
ncc: Module directory "/vercel/71ab9df5/src" attempted to require "express" but could not be resolved, assuming external.
19:34:40.881  
ncc: Module directory "/vercel/71ab9df5/src" attempted to require "celebrate" but could not be resolved, assuming external.
19:34:40.882  
ncc: Module directory "/vercel/71ab9df5/src" attempted to require "express" but could not be resolved, assuming external.
19:34:40.882  
ncc: Module directory "/vercel/71ab9df5/src" attempted to require "multer" but could not be resolved, assuming external.
19:34:40.942  
ncc: Module directory "/vercel/71ab9df5/src/config" attempted to require "multer" but could not be resolved, assuming external.
19:34:40.942  
ncc: Module directory "/vercel/71ab9df5/src/controllers" attempted to require "ip" but could not be resolved, assuming external.
19:34:40.943  
ncc: Module directory "/vercel/71ab9df5/src/controllers" attempted to require "ip" but could not be resolved, assuming external.
19:34:40.949  
ncc: Module directory "/vercel/71ab9df5/src/database" attempted to require "knex" but could not be resolved, assuming external.
19:34:41.118  
Uploading build outputs...
19:34:42.764  
Build completed. Populating build cache...
19:34:42.913  
Uploading build cache [305.00 B]...
19:34:42.968  
Build cache uploaded: 54.918ms
19:34:42.975  
Done with "src/server.js"

![image](https://user-images.githubusercontent.com/37229507/84583343-fcd5ea80-adcd-11ea-81f6-e895e4dff542.png)

**Comment by styfle** [comment, marked as answer]
[@now/node-server](https://www.npmjs.com/package/@now/node-server?activeTab=versions) was deprecated a year ago.

Instead, you should be able to deploy with [zero config](https://vercel.com/blog/zero-config) by creating a `/api/server.ts` file.

You also don't need Express because the Node.js serverless functions provide [Helpers](https://vercel.com/docs/runtimes#official-runtimes/node-js/node-js-request-and-response-objects) with similar functionality to Express.

See the migration guide here: https://vercel.com/guides/migrate-to-vercel

**Comment by sannajammeh** [comment]
This goes for every module imported, not just express. Try creating a new react app, add an API folder, and require a module. It fails with 502 once deployed to Vercel, same error as @matheusfenolio.

They work just fine in NextJS, but standalone they don't.
